### PR TITLE
Add banner with message that legacy repos are getting frozen

### PR DESCRIPTION
**Pull request by xmudrii**
This is a proposal to add a banner with the message that the legacy Google-hosted package repositories are now deprecated and will be frozen starting from September 13, 2023. The original announcement can be found here: https://k8s.io/linuxrepos

The banner is scheduled to go live from today until KubeCon NA 2023.

This change has **NOT** been discussed at all, so I'm putting this PR on hold until we don't reach consensus.
/hold

image

/assign @jeremyrickard
cc @kubernetes/release-engineering

### CODE REVIEW

File: data/announcements/scheduled.yaml
Comment (sftim) [inline data/announcements/scheduled.yaml:184]:
```suggestion
    The legacy Google-hosted package repositories (`apt.kubernetes.io` and `yum.kubernetes.io`)<br/>
```

File: data/announcements/scheduled.yaml
Comment (sftim) [inline data/announcements/scheduled.yaml:182]: - On or around 13th of September, the banner should change to a “have been frozen” message.
- I would drop the banner after September.

Comment (sftim) [inline data/announcements/scheduled.yaml:182]:
```suggestion
  title: Changes to Linux package downloads for Kubernetes
```
I'd use a blue background. People don't actually need to act urgently on this, much as we might think it's a good idea.

Comment (xmudrii) [reply data/announcements/scheduled.yaml]: Both comments are addressed, please check when you have some time

Comment (jeremyrickard) [reply data/announcements/scheduled.yaml]: I tend to agree with the importance of making the change. From an accessibility standpoint, should there be some other thing to mark it as important for anyone that has colorblindness?

Comment (jeremyrickard) [reply data/announcements/scheduled.yaml]: What did we do w/ the registry.k8s.io  update communication?

File: data/announcements/scheduled.yaml
Comment (chris-short) [inline data/announcements/scheduled.yaml:182]: Opinion
I feel like this is a little too concise to catch the right attention. Here's two other options:
"Linux packages for Kubernetes installs are changing"
"Changes to the location of Linux packages for Kubernetes"

File: data/announcements/scheduled.yaml
Comment (chris-short) [inline data/announcements/scheduled.yaml:184]: I would chop _Google-hosted_

Kubernetes is old enough and has had such rapid adoption that we should assume no one knew who was hosting anything.

The subdomains are necessary and I appreciate italicizing them.

File: data/announcements/scheduled.yaml
Comment (chris-short) [inline data/announcements/scheduled.yaml:185]: I would change _are now deprecated_ to _have been deprecated_. The community will freeze these package repositories on September 13, 2023.

**Review by chris-short — CHANGES_REQUESTED** [review]
Great work, thank you for putting this together. I made some suggested edits. They are suggestions so feel free to riff on them how you see fit.

File: data/announcements/scheduled.yaml
Comment (xmudrii) [reply data/announcements/scheduled.yaml]: > However, I am wary of using the color too frequently. Overuse could get people weary of it, and this isn't a “fix your cluster now” announcement.

I tend to agree with this, we had the red announcement quite a few times this year, but I generally hope that we will not have to use the red announcement in the future that much or at all. While this change doesn't require action _right away_, it's **a very important change**. There are some additional aspects to this:

- It might be easy to miss this change because the legacy repositories will continue working for the foreseeable future, it's just that we'll stop publishing new versions to these repositories
- Users might just think that their version went EOL or something like that, rather than that the legacy repositories are deprecated
- We don't want to make users think that "this is something that can be easily changed in the future when needed". It's very important to migrate as soon as possible. Not being able to upgrade to a newer version can make you vulnerable very quickly, and in some cases, it might not be easy to migrate _that_ quickly

File: data/announcements/scheduled.yaml
Comment (jeremyrickard) [reply data/announcements/scheduled.yaml:182]: +1 to "Changes to the location of Linux packages for Kubernetes"

File: data/announcements/scheduled.yaml
Comment (jeremyrickard) [reply data/announcements/scheduled.yaml]: > and this isn't a “fix your cluster now” announcement.

it's not an imminent, your cluster is broken sort of thing, but there will very likely be a non-trivial number of people that run into difficulties adapting to this, especially if they wait until it's time to do their next upgrade and they are skipping patch versions and land on one that will no longer be available. What about a yellow/orange background?

File: data/announcements/scheduled.yaml
Comment (jeremyrickard) [inline data/announcements/scheduled.yaml:194]: I'd also vote in favor of dropping Google-hosted. It's a detail that most people won't understand.

File: data/announcements/scheduled.yaml
Comment (xmudrii) [reply data/announcements/scheduled.yaml:182]: Initially it was "The legacy package repositories will be frozen starting with September 13, 2023" which is kind of too long and maybe we can do something like "The legacy Linux package repositories are getting frozen" as well. I don't have strong opinions.

File: data/announcements/scheduled.yaml
Comment (xmudrii) [reply data/announcements/scheduled.yaml]: Dropped in the latest commit

File: data/announcements/scheduled.yaml
Comment (xmudrii) [reply data/announcements/scheduled.yaml]: Here's some experiment with orange background (`#d95e00`):

image

Comment (xmudrii) [reply data/announcements/scheduled.yaml]: @jeremyrickard @sftim I pushed this change so that we can see how it looks like in preview.

File: data/announcements/scheduled.yaml
Comment (xmudrii) [reply data/announcements/scheduled.yaml]: Changed to "Changes to the location of Linux packages for Kubernetes"

**Comment by sftim** [comment]
Once this has LGTM, it's handy that you then squash it down to 1 commit @xmudrii. Not essential but a nice-to-have.

(And then Prow keeps the LGTM, and you unhold, and it merges)

File: data/announcements/scheduled.yaml
Comment (palnabarun) [reply data/announcements/scheduled.yaml]: I like the idea of an orange banner because:
- red has been used too many times and the action is not a do it right now kind
- blue doesn't convey the idea of long term urgency

**Review by palnabarun — APPROVED** [review]
/lgtm
/approve
(with the steering hat)

/hold for steering consensus

**Review by justaugustus — APPROVED** [review]
/lgtm
/approve (Steering & SIG Release)

**Comment by xmudrii** [comment]
@sftim Squashed all the commits into one

**Comment by mrbobbytables** [comment]
/approve
 with steering hat

**Comment by tpepper** [comment]
/approve
(Steering)

**Review by justaugustus — APPROVED** [review]
We've got a majority from Steering!
/lgtm
/hold cancel

**Comment by palnabarun** [comment]
Steering consensus is achieved with 5 votes from (@palnabarun, @justaugustus, @mrbobbytables, @cblecker, @tpepper).

**Review by BenTheElder — COMMENTED** [review]
/lgtm
/approve

**Comment by xmudrii** [comment]
Thanks a lot everyone!

**Pull request merged**
